packages/nouns-webapp/src/components/ExploreGrid/index.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState, useEffect } from 'react';
+import classes from './ExploreGrid.module.css';
+import cx from 'classnames';
+import ExploreGridItem from './ExploreGridItem';
+import Placeholder from 'react-bootstrap/esm/Placeholder';
+
+interface ExploreGridProps {
+  nounCount: number;
+  activeNoun: number;
+  selectedNoun: number | undefined;
+  setActiveNoun: Function;
+  setSelectedNoun: Function;
+  setNounsList: Function;
+  handleFocusNoun: Function;
+  isNounHoverDisabled: boolean;
+  nounsList: Noun[];
+  sortOrder: string;
+  buttonsRef: React.MutableRefObject<(HTMLButtonElement | null)[]>;
+}
+
+// noun.pics object
+type NounPic = {
+  id: number | null;
+  svg: string | undefined;
+};
+
+type Noun = {
+  id: number | null;
+  imgSrc: string | undefined;
+};
+
+const ExploreGrid: React.FC<ExploreGridProps> = props => {
+  const [individualNouns, setIndividualNouns] = useState<Noun[]>([]);
+  const placeholderNoun: Noun = { id: null, imgSrc: undefined };
+
+  // Handle events
+  const getInitialNouns = (individualCount: number) => {
+    // Fetch initial nouns by url
+    const individualNouns = new Array(individualCount)
+      .fill(placeholderNoun)
+      .map((x, i): Noun => {
+        return {
+          id: i + (props.nounCount - individualCount),
+          imgSrc: `https://noun.pics/${i + (props.nounCount - individualCount)}.svg`,
+        };
+      })
+      .reverse();
+
+    // Add initial nouns to end of placeholder array to display them first on load
+    props.setNounsList((arr: Noun[]) => [...individualNouns, ...arr]);
+    setIndividualNouns(individualNouns);
+    // After initial nouns are set, run range calls
+    rangeCalls(props.nounCount, individualNouns);
+  };
+
+  // Range calls
+  const initialChunkSize = 10;
+  const rangeChunkSize = 100;
+  const rangeCalls = async (nounCount: number, individualNouns: Noun[]) => {
+    if (nounCount >= 0) {
+      for (let i = nounCount - individualNouns.length; i >= 0; i -= rangeChunkSize) {
+        const start = i - rangeChunkSize < 0 ? 0 : i - rangeChunkSize;
+        const end = i - 1;
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const nounsRange = await fetchNouns(start, end);
+      }
+    }
+  };
+
+  const fetchNouns = async (start: number, end: number) => {
+    const url = `https://noun.pics/range?start=${start}&end=${end}`;
```

Comment:
These external calls could be avoided as we already have enough local information to construct existing Nouns.

That is, the cached Nouns seeds, which are indexed by Noun ID, and the RLE `ImageData`.

Reply:
iirc, that was actually the original approach here (using `StandaloneNounImage`) but with so many images on one page it felt slow to interact with, so we opted for static images via noun.pics.

Reply:
Interesting, wouldn't expect the image generation to add much overhead. Parsing the base64 image data is probably more CPU-intensive, but both solutions are bound by that. Maybe due to `StandaloneNounImage`, which should be trimmed down if using here.

Regardless, can push it off and possibly update later.

Reply:
i'll see if i can dig up a deploy preview link for us to compare against. and in the meantime i'll mark this this issue resolved for now.